sdk/containerregistry/container-registry/review/container-registry.api.md

```diff
@@ -4,41 +4,73 @@
 
 ```ts
 
+import * as coreClient from '@azure/core-client';
+import * as coreHttps from '@azure/core-rest-pipeline';
 import { OperationOptions } from '@azure/core-client';
 import { PagedAsyncIterableIterator } from '@azure/core-paging';
 import { PipelineOptions } from '@azure/core-rest-pipeline';
 import { TokenCredential } from '@azure/core-auth';
 
+// @public
+export type ArtifactArchitecture = "386" | "amd64" | "arm" | "arm64" | "mips" | "mipsle" | "mips64" | "mips64le" | "ppc64" | "ppc64le" | "riscv64" | "s390x" | "wasm" | string;
+
+// @public
+export interface ArtifactManifestProperties {
+    readonly architecture?: ArtifactArchitecture | null;
+    readonly createdOn?: Date;
+    readonly digest?: string;
+    readonly lastUpdatedOn?: Date;
+    manifests?: ArtifactManifestProperties[];
+    readonly operatingSystem?: ArtifactOperatingSystem | null;
+    readonly repositoryName?: string;
+    readonly size?: number;
+    readonly tags?: string[];
+    readonly writeableProperties?: ContentProperties;
+}
+
+// @public
+export type ArtifactOperatingSystem = "aix" | "android" | "darwin" | "dragonfly" | "freebsd" | "illumos" | "ios" | "js" | "linux" | "netbsd" | "openbsd" | "plan9" | "solaris" | "windows" | string;
+
+// @public
+export interface ArtifactTagProperties {
+    readonly createdOn: Date;
+    readonly digest: string;
+    readonly lastUpdatedOn: Date;
+    readonly name: string;
+    readonly repository: string;
+    readonly writeableProperties: ContentProperties;
+}
+
 // @public
 export class ContainerRegistryClient {
     constructor(endpointUrl: string, credential: TokenCredential, options?: ContainerRegistryClientOptions);
-    deleteRepository(name: string, options?: DeleteRepositoryOptions): Promise<DeleteRepositoryResult>;
-    endpoint: string;
-    getRepositoryClient(repository: string): ContainerRepositoryClient;
-    listRepositories(options?: ListRepositoriesOptions): PagedAsyncIterableIterator<string, string[]>;
-    }
+    deleteRepository(repositoryName: string, options?: DeleteRepositoryOptions): Promise<DeleteRepositoryResult>;
+    getArtifact(repositoryName: string, tagOrDigest: string): RegistryArtifact;
+    getRepository(repositoryName: string): ContainerRepository;
+    listRepositoryNames(options?: ListRepositoriesOptions): PagedAsyncIterableIterator<string, string[]>;
+    loginServer: string;
+    name: string;
+    registryUrl: string;
+}
 
 // @public
 export interface ContainerRegistryClientOptions extends PipelineOptions {
 }
 
 // @public
-export class ContainerRepositoryClient {
-    constructor(endpointUrl: string, repository: string, credential: TokenCredential, options?: ContainerRegistryClientOptions);
+export class ContainerRepository {
+    // Warning: (ae-forgotten-export) The symbol "GeneratedClient" needs to be exported by the entry point index.d.ts
```

Comment:
this feels a bit leaky, we should be able to avoid exposing `GeneratedClient` -- seems like this class should be hidden and we can expose a compatible `ContainerRegistry` interface that we return the class instance for